chrome/content/finger.js

```diff
@@ -115,4 +115,21 @@ var otrFinger = {
     fingerTreeView.selection.selectEventsSuppressed = false;
   },
 
+  add: function() {
+    fingerTreeView.selection.selectEventsSuppressed = true;
+    const nsIFilePicker = Ci.nsIFilePicker;
+    let fp = Cc["@mozilla.org/filepicker;1"].createInstance(nsIFilePicker);
+    fp.init(window, _("openfile.label"), nsIFilePicker.modeOpen);
+    fp.appendFilters(nsIFilePicker.filterAll | nsIFilePicker.filterText);
+    let rv = fp.show();
+    if (rv == nsIFilePicker.returnOK) {
+      otr.readFingerprints(fp.file.path);
+      otr.writeFingerprints();
+      fingers = otr.knownFingerprints();
+      fingerTreeView.rowCount = fingers.length;
+      fingerTree.view = fingerTreeView;
```

Comment:
Maybe factor out these three lines since they're used in `onload` above as well.